core/domain/html_cleaner.py

```diff
@@ -192,3 +193,401 @@ def get_rte_components(html_string: str) -> List[ComponentsDict]:
+        ValidationError. Collapsible content-with-value attribute is not
+            present.
+        ValidationError. Collapsible heading-with-value attribute is not
+            present.
+        ValidationError. Collapsible heading-with-value attribute is empty.
+    """
+    soup, empty_values = _initialize_bs4_and_empty_values_variable(html_data)
+    tabs_tags = soup.find_all('oppia-noninteractive-tabs')
+    for tag in tabs_tags:
+        if tag.has_attr('tab_contents-with-value'):
+            tab_content_json = utils.unescape_html(
+                tag['tab_contents-with-value'])
+            tab_content_list = json.loads(tab_content_json)
+            if len(tab_content_list) == 0:
+                raise utils.ValidationError(
+                    'No tabs are present inside the tabs tag.'
+                )
+
+            for tab_content in tab_content_list:
+                if 'title' not in tab_content:
+                    raise utils.ValidationError(
+                        'No title attribute is present inside the tabs tag.'
+                    )
+
+                if tab_content['title'].strip() in empty_values:
+                    raise utils.ValidationError(
+                        'Title present inside tabs tag is empty.'
+                    )
+
+                if 'content' not in tab_content:
+                    raise utils.ValidationError(
+                        'No content attribute is present inside the tabs tag.'
+                    )
+
+                if tab_content['content'].strip() in empty_values:
+                    raise utils.ValidationError(
+                        'Content present inside tabs tag is empty.'
+                    )
+
+                validate_rte_tags(
+                    tab_content['content'],
+                    is_tag_nested_inside_tabs_or_collapsible=True
+                )
+        else:
+            raise utils.ValidationError(
+                'No content attribute is present inside the tabs tag.'
+            )
+
+    collapsibles_tags = soup.find_all('oppia-noninteractive-collapsible')
+    for tag in collapsibles_tags:
+        if tag.has_attr('content-with-value'):
+            collapsible_content_json = (
+                utils.unescape_html(tag['content-with-value'])
+            )
+            collapsible_content = json.loads(
+                collapsible_content_json).replace('\\"', '')
+            if collapsible_content.strip() in empty_values:
+                raise utils.ValidationError(
+                    'No collapsible content is present inside the tag.'
+                )
+
+            validate_rte_tags(
+                collapsible_content,
+                is_tag_nested_inside_tabs_or_collapsible=True
+            )
+        else:
+            raise utils.ValidationError(
+                'No content attribute present in collapsible tag.'
+            )
+
+        if tag.has_attr('heading-with-value'):
```

Comment:
do `if not xxx: raise` instead of `else: raise`

Reply:
ditto elsewhere

Reply:
Done